bpf/generictracer/protocol_tcp.h

```diff
@@ -187,7 +193,8 @@ static __always_inline void handle_unknown_tcp_connection(pid_connection_info_t
                                                           u16 orig_dport,
                                                           enum protocol_type protocol_type) {
     tcp_req_t *existing = bpf_map_lookup_elem(&ongoing_tcp_req, pid_conn);
-
+    u32 netns = task_netns();
+    bool is_server = is_listening(pid_conn->conn.d_port, netns);
```

Comment:
This check could misclassify a client as a server in the case that some unrelated service happens to be bound to this process's port (such as a metrics server). Not sure how likely that is, but it could be worth having a more robust classification here?

Reply:
In the first implementation, the algorithm tried to parse the event as a request and then as a response because there was a bug and `is_server` had the wrong values.

The `is_listening` function is used in the [infer_packet_type](https://github.com/open-telemetry/opentelemetry-ebpf-instrumentation/blob/3781840a42ffed54dbeb070aa22ec72799ed61b0/bpf/generictracer/protocol_common.h#L44), shouldn't we have the same problem there too?

Reply:
I think is worth commenting this in the code so if at some point in the future this issue emerges, someone trying to look for the cause can have more context here.